### ISSUE TITLE
Help: Landing with SL GHSTACK failed halfway in private repo

### ISSUE THREAD

**Issue by tangentlin** [facebook/sapling#609]
I have no issue landing Sapling PR's in private repository for some time, but lately landing usually crashed half way such that the PR is marked merged, but the content has not been merged into `main`.  I am admin to the private repo and can confirm the repo allows forced push.  Any pointers on where else I need to look into?

```Shell
error: failed to push some refs to 'ssh://github.com/XYZPrivate/abc.git'
** Sapling SCM (version 0.2.20230330-193452+h69692651) has crashed:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/__init__.py", line 106, in run
    dispatch.run(args, fin, fout, ferr, config)
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 161, in run
    status = (dispatch(req) or 0) & 255
              ^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 499, in dispatch
    ret = _runcatch(req)
          ^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 687, in _runcatch
    return _callcatch(ui, req, _runcatchfunc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 696, in _callcatch
    return scmutil.callcatch(ui, req, func)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/scmutil.py", line 143, in callcatch
    return func()
           ^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 670, in _runcatchfunc
    return _dispatch(req)
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 1197, in _dispatch
    ret = runcommand(
          ^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/ext/undo.py", line 148, in _runcommandwrapper
    result = orig(lui, repo, cmd, fullargs, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/ext/journal.py", line 80, in runcommand
    return orig(lui, repo, cmd, fullargs, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/ext/sparse.py", line 478, in _tracktelemetry
    res = runcommand(lui, repo, *args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/ext/copytrace.py", line 164, in _runcommand
    return orig(lui, repo, cmd, fullargs, ui, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 886, in runcommand
    ret = _runcommand(ui, options, cmd, d)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 1238, in _runcommand
    return cmdfunc()
           ^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/dispatch.py", line 1196, in <lambda>
    d = lambda: util.checksignature(func)(ui, *args, **strcmdopt)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/util.py", line 1374, in check
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/edenscm/ext/ghstack/__init__.py", line 149, in land_cmd
    ghstack.land.main(
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ghstack/land.py", line 23, in main
    closed
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ghstack/sapling_land.py", line 163, in main
    sh.run_sapling_command("push", "--rev", push_rev, "--to", default_branch)
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ghstack/sapling_shell.py", line 117, in run_sapling_command
    return self._run_sapling_command(list(args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ghstack/sapling_shell.py", line 123, in _run_sapling_command
    stdout = self.sh(*full_args, env=env)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ghstack/shell.py", line 234, in sh
RuntimeError: /opt/homebrew/bin/sl push --rev bf637eae0a11f7a987214242a20d2e6df46cec8f --to main failed with exit code 1. Output was 'b'''
```

**Comment by tangentlin** [comment]
Some suggested to run a debug `sl push --rev bf637eae0a11f7a987214242a20d2e6df46cec8f --to main --debug`, and below is the result.  

```Shell
watchman sockpath is set as /opt/facebook/watchman/var/run/watchman/tangent.lin-state/sock
Pushing to ssh://github.com/XYZCompany/abc.git
Enumerating objects: 77, done.
Counting objects: 100% (77/77), done.
Delta compression using up to 12 threads
Compressing objects: 100% (43/43), done.
Writing objects: 100% (53/53), 5.86 KiB | 2.93 MiB/s, done.
Total 53 (delta 30), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (30/30), completed with 18 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: Waiting on code owner review from premisedata/glados. 3 of 3 required status checks are expected.
To ssh://github.com/premisedata/glados.git
 ! [remote rejected]   bf637eae0a11f7a987214242a20d2e6df46cec8f -> main (protected branch hook declined)
error: failed to push some refs to 'ssh://github.com/XYZCompany/abc.git'
```
